services/api/src/resources/task/resolvers.ts

```diff
@@ -419,7 +419,8 @@ export const taskDrushSqlDump: ResolverFn = async (
     project: envPerm.project
   });
 
-  const command = String.raw`file="/tmp/$LAGOON_PROJECT-$LAGOON_GIT_SAFE_BRANCH-$(date --iso-8601=seconds).sql" && drush sql-dump --result-file=$file --gzip && \
+  const command = String.raw`file="/tmp/$LAGOON_PROJECT-$LAGOON_GIT_SAFE_BRANCH-$(date --iso-8601=seconds).sql" && DRUSH_MAJOR_VERSION=$(drush status --field=drupal-version | grep -oE '^\s*[0-9]+') && \
+if [[ $DRUSH_MAJOR_VERSION -gt 9 ]]; then drush sql-dump --extra-dump=--no-tablespaces --result-file=$file --gzip; else drush sql-dump --extra=--no-tablespaces --result-file=$file --gzip; fi && \
```

Comment:
Should this not be `ge` to be 9 or greater instead of greater than 9?

```suggestion
if [[ $DRUSH_MAJOR_VERSION -ge 9 ]]; then drush sql-dump --extra-dump=--no-tablespaces --result-file=$file --gzip; else drush sql-dump --extra=--no-tablespaces --result-file=$file --gzip; fi && \
```

Reply:
Ah yeah - that's right, it seems drush 9 introduced `--extra-dump`

drush 10 = https://www.drush.org/latest/commands/sql_dump/
drush 9 = https://drushcommands.com/drush-9x/sql/sql:dump/
drush 8 = https://drushcommands.com/drush-8x/sql/sql-dump/